packages/flutter_tools/lib/src/commands/create_base.dart

```diff
@@ -18,11 +18,27 @@ import '../project.dart';
 import '../runner/flutter_command.dart';
 import '../template.dart';
 
-const _kAvailablePlatforms = <String>['ios', 'android', 'windows', 'linux', 'macos', 'web'];
+const _kAvailablePlatforms = <String>[
+  'ios',
+  'android',
+  'windows',
+  'linux',
+  'macos',
+  'web',
+  'darwin',
+];
```

Comment:
`_kAvailablePlatforms` is used to show the `--platform` options when you do `flutter create --help`. 

`flutter create` can be used to create different types of projects (apps, modules, packages, plugins, etc). The `darwin` platform is only supported for creating plugins, though. I think maybe you can set `allowedHelp` in `addPlatformsOptions` to give it a custom description. Maybe something like "only support for plugins"